### PR TITLE
Refactoring service retirement start_retirement method.

### PR DESCRIPTION
**Pull request by billfitzgerald0120**
Refactoring Service/Retirement/StateMachines/Methods.class/methods/start_retirement.rb method with spec. This PR is based on the issue below.

https://github.com/ManageIQ/manageiq-content/issues/8

@miq-bot add_label refactoring

### CODE REVIEW

File: spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      current_object.parent = root_object
+      service.object = current_object
+    end
+  end
+
+  let(:svc_model_request) do
+    MiqAeMethodService::MiqAeServiceServiceRetireRequest.find(request.id)
+  end
+
+  it "#start_retirement is nil and service is retiring" do
```
Comment (lfu) [inline spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb]: please make this more descriptive.

File: spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    expect(service.retired?).to be_falsey
+  end
+
+  it "#retired? - true" do
+    service.retirement_state = 'retired'
+    service.retired = true
+    expect(service.retired?).to be_truthy
+  end
+
+  it "starts retirement" do
```
Comment (lfu) [inline spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb]: What does this test?

File: content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement.rb [hunk 3]
```diff
@@ -1,26 +1,46 @@
...
-end
+              private
 
-if service.retiring?
-  $evm.log('error', "Service is in the process of being retired. Aborting current State Machine.")
-  exit MIQ_ABORT
+              def service
+                @service ||= @handle.root["service"].tap do |service|
+                  if service.nil?
+                    @handle.log(:error, 'Service Object not found')
```
Comment (lfu) [inline content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement.rb]: Seems perfect places to use ```ManageIQ::Automate::System::CommonMethods::Utils::LogObject.log_and_raise```.

Comment (billfitzgerald0120) [reply content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement.rb]: I want to do this in a followup PR

File: spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb (hunk 2 above)
Comment (lfu) [reply spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb]: This does not test anything.

Comment (billfitzgerald0120) [reply spec/content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/start_retirement_spec.rb]: I'm trying to test the good condition.  Retirement_state is nil and then I start retirement and the retirement_state is equal to retiring.

**Comment by billfitzgerald0120** [comment]
Closing this PR, github.com/ManageIQ/manageiq-content/pull/601 replaces this PR.

**Pull request closed without merging**